### PR TITLE
Cleaned up and improved the profiling code

### PR DESCRIPTION
**Pull request by ceriottm**
Already used it to tweak a few bottlenecks, although actually there doesn't seem to be many. 
Also took the opportunity to polish a few examples, but would be probably better to do this in a separate branch so I'm stopping.

### CODE REVIEW

File: bin/i-pi [hunk 1]
```diff
@@ -83,7 +83,14 @@ if __name__ == '__main__':
 
     parser.add_option('-p', '--profile',
                       action='store_true', dest='do_yappi', default=False,
-                      help='Profile this run using Yappi.')
+                      help='Profile this run using yappi.')
+
+    parser.add_option('--profiler-clock', dest='yappi_clock', default="cpu",
+                      choices=['cpu', 'wall'],
+                      help='Type of profiling timing.')
```
Comment (mahrossi) [inline bin/i-pi:90]: Change to more descriptive? e.g. Profiling timing from cpu time or wallclock time.

File: examples/features/Classical_molecular_dynamics/NPzT_ensemble_BZP/lammps_h2o_bulk_Ih/simulation.xc.pdb
Comment (mahrossi) [inline examples/features/Classical_molecular_dynamics/NPzT_ensemble_BZP/lammps_h2o_bulk_Ih/simulation.xc.pdb:1]: Should this pdb file really be here? I think not

**Review by mahrossi — CHANGES_REQUESTED** [review]
Only minor, looks good overall. I would only require an example on how to run with profiling OR even better, an entry in the manual on how to profile the code.

**Comment by venkatkapil24** [comment]
I am back from holiday, so I am available to take a look. But it would be great to have an example showing how to profile! Do we need a utility directory in examples?

File: examples/features/Classical_molecular_dynamics/NPzT_ensemble_BZP/lammps_h2o_bulk_Ih/simulation.xc.pdb
Comment (ceriottm) [reply examples/features/Classical_molecular_dynamics/NPzT_ensemble_BZP/lammps_h2o_bulk_Ih/simulation.xc.pdb:1]: No def. not.

**Comment by ceriottm** [comment]
> I am back from holiday, so I am available to take a look. But it would be great to have an example showing how to profile! Do we need a utility directory in examples?

If you can add an example, that'd be great. I think it could belong to the `hpc_scripts` folder.  Let's discuss examples structure in a separate PR, there are several things I'd like to change there

File: doc/src/troubleshooting.rst [hunk 2]
```diff
@@ -278,3 +278,27 @@ Mathematical errors
...
+-  *reduce I/O*: outputting hundreds of beads configurations at each time
+   step is going to be slow in any scenario, but particularly so when
+   using text files and Python. Reduce the output frequency using a larger
+   `stride`, and/or `flush` less often. 
+-  *profile i-PI*: if you still think i-PI is being unreasonably slow,
+   you can contact the developers - your setup might have revealed some 
+   kind of bottleneck. It will help us if you can also generate a profiler
+   output from your run (possibly with only a small number of steps). 
+   You can generate a profiler log by running with the `-p` option. 
+   You will need to install the `yappi` profiler.
```
Comment (mahrossi) [inline doc/src/troubleshooting.rst:304]: This is a great addition to the manual. Sorry to be pedantic, but for the user the actual command is needed. I thought one had to run `yappi i-pi ... `, but you suggest python -p i-pi? Just writing the command helps. then I would merge.

**Pull request merged**